salt/hostname/init.sls

```diff
@@ -1,13 +1,22 @@
-{% set hostname = grains['id'] + '.' + pillar['internal_infra_domain'] %}
+caasp_fqdn:
+  grains.present:
+    - value: {{ grains['id'] }}.{{ pillar['internal_infra_domain'] }}
 
+# Both of the below are due to be removed, and can't use the `caasp_fqdn` grain
+# as it's not available in grains until the next state, so lets leave them as is
+# for the moment
 /etc/hostname:
   file.managed:
-    - contents: {{ hostname }}
+    - contents: {{ grains['id'] }}.{{ pillar['internal_infra_domain'] }}
```

Comment:
Yea, I really wanted a "non-runtime" way to set this, but that didn't seem possible